explorer/syntax/parser.ypp

```diff
@@ -300,148 +300,148 @@
-        context.RecordSyntaxError(
-            llvm::formatv("Invalid type literal: {0}", $1));
+      if (!llvm::to_integer(llvm::StringRef($sized_type_literal).substr(1), val)) {
+        pl_context.RecordSyntaxError(
+            llvm::formatv("Invalid type literal: {0}", $sized_type_literal));
         YYERROR;
-      } else if ($1[0] != 'i' || val != 32) {
-        context.RecordSyntaxError(
-            llvm::formatv("Only i32 is supported for now: {0}", $1));
+      } else if ($sized_type_literal[0] != 'i' || val != 32) {
+        pl_context.RecordSyntaxError(
+            llvm::formatv("Only i32 is supported for now: {0}", $sized_type_literal));
         YYERROR;
       } else {
-        $$ = arena->New<IntTypeLiteral>(context.source_loc());
+        $primary_expression = arena->New<IntTypeLiteral>(pl_context.source_loc());
       }
     }
 | SELF
     // TODO: Should we create a new TypeLiteral for `Self`?
-    { $$ = arena->New<IdentifierExpression>(context.source_loc(), "Self"); }
+    { $primary_expression = arena->New<IdentifierExpression>(pl_context.source_loc(), "Self"); }
 | STRING
-    { $$ = arena->New<StringTypeLiteral>(context.source_loc()); }
+    { $primary_expression = arena->New<StringTypeLiteral>(pl_context.source_loc()); }
 | BOOL
-    { $$ = arena->New<BoolTypeLiteral>(context.source_loc()); }
+    { $primary_expression = arena->New<BoolTypeLiteral>(pl_context.source_loc()); }
 | TYPE
-    { $$ = arena->New<TypeTypeLiteral>(context.source_loc()); }
+    { $primary_expression = arena->New<TypeTypeLiteral>(pl_context.source_loc()); }
 | CONTINUATION_TYPE
-    { $$ = arena->New<ContinuationTypeLiteral>(context.source_loc()); }
-| paren_expression { $$ = $1; }
-| struct_literal { $$ = $1; }
-| struct_type_literal { $$ = $1; }
-| LEFT_SQUARE_BRACKET expression SEMICOLON expression RIGHT_SQUARE_BRACKET
-    { $$ = arena->New<ArrayTypeLiteral>(context.source_loc(), $2, $4); }
+    { $primary_expression = arena->New<ContinuationTypeLiteral>(pl_context.source_loc()); }
+| paren_expression { $primary_expression = $paren_expression; }
+| struct_literal { $primary_expression = $struct_literal; }
+| struct_type_literal { $primary_expression = $struct_type_literal; }
+| LEFT_SQUARE_BRACKET expression[exp1] SEMICOLON expression[exp2] RIGHT_SQUARE_BRACKET
+    { $primary_expression = arena->New<ArrayTypeLiteral>(pl_context.source_loc(), $exp1, $exp2); }
 ;
 postfix_expression:
   primary_expression
-| postfix_expression designator
+| postfix_expression[lhs] designator
     {
-      $$ = arena->New<SimpleMemberAccessExpression>(context.source_loc(), $1,
-                                                    $2);
+      $$ = arena->New<SimpleMemberAccessExpression>(pl_context.source_loc(), $lhs,
+                                                    $designator);
     }
-| postfix_expression PERIOD LEFT_PARENTHESIS expression RIGHT_PARENTHESIS
+| postfix_expression[lhs] PERIOD LEFT_PARENTHESIS expression RIGHT_PARENTHESIS
```

Comment:
Where I added `[some name]` is because the `some name` appears more than once in the rule and need `[another name]` to disambiguate.

Reply:
On this particular case we can not use `expression` because it's already a name used in the rule.

Reply:
Sorry, I wasn't very clear. Here's what I'm suggesting:

```suggestion
| postfix_expression[lhs] PERIOD LEFT_PARENTHESIS expression[rhs] RIGHT_PARENTHESIS
```

(and then use `rhs` instead of `expression` in the body below)

I know that's not necessary for disambiguation, but I think it will make the body clearer for the reader.